PyQgsExternalStorageWebDAV test fail (with .ci/run_tests.sh)

### What is the bug or the crash?

Running `.ci/run_tests.sh` fail at running PyQgsExternalStorageWebDAV:
```
436/813 Test #446: PyQgsExternalStorageWebDAV .........................***Failed    0.91 sec
export =
export PYTHONPATH=/root/QGIS/build-ci/output/python:/root/QGIS/build-ci/output/python/plugins:/root/QGIS/tests/src/python:
-- Running timeout -sSIGSEGV 55s /usr/bin/python3.8 /root/QGIS/tests/src/python/test_qgsexternalstorage_webdav.py
salt hex: a01b25123aad4522c1587dba66ba5781
hash hex: fe5eb79eca1668491958b2200dd8564a
cipheriv hex: 3b8ca348fba349813d03e9317eef2ef96fc49a321b04db887e924b70b3c7f167
..FF.F
======================================================================
FAIL: testStoreFetchFileImmediately (__main__.TestPyQgsExternalStorageWebDAV)
Test file storing and fetching (Immediately mode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/QGIS/tests/src/python/test_qgsexternalstorage_base.py", line 211, in testStoreFetchFileImmediately
    self.assertEqual(len(spyErrorOccurred), 0)
AssertionError: 1 != 0

======================================================================
FAIL: testStoreFetchFileLater (__main__.TestPyQgsExternalStorageWebDAV)
Test file storing and fetching (Later mode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/QGIS/tests/src/python/test_qgsexternalstorage_base.py", line 125, in testStoreFetchFileLater
    self.assertEqual(len(spyErrorOccurred), 0)
AssertionError: 1 != 0

===============================================================Application state:
QGIS_PREFIX_PATH env var:               /root/QGIS/build-ci/output
Prefix:
Plugin Path:            /root/QGIS/build-ci/output/lib/qgis/plugins
Package Data Path:      /root/QGIS/build-ci/output/data
Active Theme Name:
Active Theme Path:      /root/QGIS/build-ci/output/data/resources/themes//icons/
Default Theme Path:     :/images/themes/default/
SVG Search Paths:       /root/QGIS/build-ci/output/data/svg/
                /tmp/QGIS-PythonTestConfigPath700rxmxi/profiles/default/svg/
User DB Path:   /root/QGIS/build-ci/output/data/resources/qgis.db
Auth DB Path:   /tmp/QGIS-PythonTestConfigPath700rxmxi/profiles/default/qgis-auth.db

Network(1): Host nothinghere not found
Network(1): Error transferring http://webdav:80/webdav_tests/tmpt15ecdjn.txt - server replied: Internal Server Error
Network(1): Error transferring http://webdav:80/webdav_tests/tmp56owoif8.txt - server replied: Internal Server Error
Network(1): Host requires authentication
Network(1): Error transferring http://webdav:80/webdav_tests/tmpcjyztdi9.txt - server replied: Internal Server =======
FAIL: testStoreWithoutFileName (__main__.TestPyQgsExternalStorageWebDAV)
Test file storing and fetching
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/QGIS/tests/src/python/test_qgsexternalstorage_base.py", line 350, in testStoreWithoutFileName
    self.assertEqual(len(spyErrorOccurred), 0)
AssertionError: 1 != 0
```

### Steps to reproduce the issue

Run `.ci/run_tests.sh` from the root source directory of QGIS (won't work in a wortree, make sure .git is a directory directory)

### Versions

Current master branch as of May 23, 2022 - ab524fc5e1039804c26dcc06ccb9e178d97e92aa

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

--- a/.ci/run_tests.sh
+++ b/.ci/run_tests.sh
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+set -e
+
 # check for docker-compose and docker availability
 command -v docker > /dev/null || {
 	echo "Please install docker" >&2
@@ -120,6 +122,9 @@ else
   echo "--=[ Starting interactive shell into test environment"
   COMMAND=bash
 fi
+
+# Create an empty webdav folder with appropriate permissions so www user can write inside it
+mkdir -p /tmp/webdav_tests && chmod 777 /tmp/webdav_tests
 
 docker-compose \
   -f .docker/docker-compose-testing.yml \

